src/main/scala/mirroring/services/writer/ChangeTrackingService.scala

```diff
@@ -99,4 +111,50 @@ class ChangeTrackingService(
   override def getUserMetadataJSON: String = {
     userMetadataJSON
   }
+
+  private def deleteRows(data: DataFrame): Unit = {
+    val keys: Array[Column] =
+      keysCombined.map(x => new Column(s"${Config.SourceAlias}.$sourceColPrefix$x"))
+    var deleteOperations = data
+      .as(Config.SourceAlias)
+      .where(deleteCondition)
+      .select(keys: _*)
+
+    for (name <- deleteOperations.columns) {
+      deleteOperations = deleteOperations.withColumnRenamed(name, name.replace(sourceColPrefix, ""))
+    }
```

Comment:
Maybe make sense to use `new Column("someColumn").as("someAlias")` in this case?

Reply:
Removed completely.